src/helpers/tagUtils.ts

```diff
@@ -0,0 +1,163 @@
+import { db } from '../database/index.js';
+import type { Tag } from '@prisma/client';
+
+/**
+ * Creates a new tag for a guild.
+ * @param guildId Discord guild ID
+ * @param name Tag name
+ * @param content Tag content (URL or text)
+ * @param createdBy Discord user ID of creator
+ * @returns The created tag
+ * @throws Error if tag with that name already exists
+ */
+export async function createTag(
+	guildId: string,
+	name: string,
+	content: string,
+	createdBy: string
+): Promise<Tag> {
+	// Check if tag already exists
+	const existing = await db.tag.findUnique({
+		where: {
+			guild_tag: {
+				guildId,
+				name: name.toLowerCase(),
+			},
+		},
+	});
+
+	if (existing) {
+		throw new Error(`A tag named '${name}' already exists in this server.`);
+	}
+
+	return db.tag.create({
+		data: {
+			guildId,
+			name: name.toLowerCase(),
+			content,
```

Comment:
Rather than storing the tag content URL as-is, we should probably download the image data and store it somewhere (in the database or some file) to send later. This way we aren't relying on ephemeral hotlinks that may disappear or expire later at any time.

I'm not sure about text tags, but if we want arbitrary strings as well, then we can do both probably, somehow.

Reply:
There is a way to launch a modal interaction with a file upload option, which might be useful: https://discordjs.guide/legacy/interactions/modals#file-upload

For example, `/tag add` could take no options, and instead launch a modal with a name field and a file input, as well as a field for alt text